crates/frontend/src/pages/config_version_list.rs

```diff
@@ -0,0 +1,206 @@
+                                Value::String(config_version.id.to_string()),
+                            );
+                            map.insert(
+                                "created_at".to_string(),
+                                json!(config_version.created_at),
+                            );
+                            map.insert("tags".to_string(), json!(config_version.tags));
+                            map
+                        })
+                        .collect(),
+                    Err(_) => vec![],
+                }
+            },
+        );
+
+    let handle_next_click = Callback::new(move |total_pages: i64| {
+        set_filters.update(|f| {
+            f.page = match f.page {
+                Some(p) if p < total_pages => Some(p + 1),
+                Some(p) => Some(p),
+                None => Some(1),
+            }
+        });
+    });
+
+    let handle_prev_click = Callback::new(move |_| {
+        set_filters.update(|f| {
+            f.page = match f.page {
+                Some(p) if p > 1 => Some(p - 1),
+                Some(_) => Some(1),
+                None => Some(1),
+            }
+        });
+    });
+
+    view! {
+        <div class="p-8">
+            <Suspense fallback=move || view! { <Skeleton/> }>
+                <div class="pb-4">
+                    {move || {
+                        let snapshot_res = snapshots_resource.get();
+                        let total_items = match snapshot_res {
+                            Some(snapshot_resp) => snapshot_resp.len().to_string(),
+                            _ => "0".to_string(),
+                        };
+                        view! {
+                            <Stat
+                                heading="ConfigVersions"
+                                icon="ri-camera-lens-fill"
+                                number=total_items
+                            />
+                        }
+                    }}
+
+                </div>
+                <div class="card rounded-xl w-full bg-base-100 shadow">
+                    <div class="card-body">
+                        <div class="flex justify-between">
+                            <h2 class="card-title">Config Versions</h2>
+                        </div>
+                        <div>
+                            {move || {
+                                let value = snapshots_resource.get();
+                                let filters = filters.get();
+                                match value {
+                                    Some(snapshot_resp) => {
+                                        let page = filters.page.unwrap_or(1);
+                                        let count = filters.count.unwrap_or(10);
+                                        let total_items = snapshot_resp.len() as i64;
+                                        let total_pages = (total_items as f64 / count as f64).ceil()
+                                            as i64;
```

Comment:
Can you use the api response's `total_pages` instead of calculating again, and this is not correct as `total_items` is `length` of the versions list which is `10`, so everytime you will only have one single page.